src/components/FilterPanel.tsx

```diff
@@ -44,6 +45,7 @@ export default function FilterPanel({
   const [grades, setGrades] = useState(params.grades);
   const [majors, setMajors] = useState(params.majors);
   const [states, setStates] = useState(params.states);
+  const [schools, setSchools] = useState(params.schools);
   const location = useLocation();
 
   useEffect(() => {
```

Comment:
Could you remind me why we have this effect? This really seems like something `useQueryParams` should be doing. I'm surprised we haven't filed a bug to address this.

Reply:
so the filters dont get applied right away when the user choose them.